x/harvest/keeper/borrow.go

```diff
@@ -32,3 +38,60 @@ func (k Keeper) Borrow(ctx sdk.Context, borrower sdk.AccAddress, coins sdk.Coins
 
 	return nil
 }
+
+// ValidateBorrow validates a borrow request against borrower and protocol requirements
+func (k Keeper) ValidateBorrow(ctx sdk.Context, borrower sdk.AccAddress, amount sdk.Coin) error {
+	proposedBorrowPrice, proposedBorrowConversionFactor, err := k.getAssetPrice(ctx, amount.Denom)
```

Comment:
Included this suggestion in https://github.com/Kava-Labs/kava/pull/710/commits/e0291e46b8a73c97571885dc8d591e2d7e787baa. Since param fetching adds overhead I agree that this is the right choice, even if lines 44-52 duplicate code from `calculateUSDValue`.